### PR TITLE
WIP: compound v3 asset proxy

### PR DESCRIPTION
**Pull request by rahul-kothari**
Hi! I am working on a fixed borrow protocol at component.fi.

I am investigating how to add an asset proxy of compound III into element v1. The problem is that compound III acts like Aave in that cToken balance keeps increasing with time. So `deposit()` or `underlying()` doesn't work the same as v2 of compound. 

Currently, I have just written the asset proxy. Will add tests later, but wanted some eyes on if I am going in the right direction. 

Basically added `yieldSharesIssued` which keeps track of tokens issued by the deposit() so that we can later calculate `underlying()` correctly! 
Shoutout to @Padraic-O-Mhuiris for the idea

### CODE REVIEW

**Comment by Padraic-O-Mhuiris** [comment]
This needs some integration-level testing validating the withdrawals and deposits from compound are being made correctly.
